jax/interpreters/partial_eval.py

```diff
@@ -511,6 +513,12 @@ def partial_eval_wrapper_nounits(
 call_partial_eval_rules: Dict[Primitive, Callable] = {}
 call_param_updaters: Dict[Primitive, Callable] = {}
 
+def _closed_call_param_updater(params, _, __):
+  jaxpr = params.get('call_jaxpr')
+  if jaxpr is None: return params
```

Comment:
When is jaxpr None here?

Reply:
Yeah good question, I forget... let me see if I can exercise this.

Reply:
Ah, it's because in `JaxprTrace.process_call` we actually call [the same call_param_updater](https://github.com/google/jax/blob/ba0a2b3a984858162c6465366ca9fdcc1c97a1c8/jax/interpreters/partial_eval.py#L196) for both the [bind-form](https://github.com/google/jax/blob/ba0a2b3a984858162c6465366ca9fdcc1c97a1c8/jax/interpreters/partial_eval.py#L210) and [jaxpr-form](https://github.com/google/jax/blob/ba0a2b3a984858162c6465366ca9fdcc1c97a1c8/jax/interpreters/partial_eval.py#L225) parameter versions. Usually it's just used to update params like `donated_invars`, and it doesn't matter whether we're working with the bind-form or the jaxpr-form (e.g. [for xla_call](https://github.com/google/jax/blob/main/jax/interpreters/xla.py#L403-L415)).

Reply:
Makes sense, thanks!